### PR TITLE
Add tests for consul service discovery

### PR DESCRIPTION
**Pull request by neumayer**

### CODE REVIEW

**Comment by beorn7** [comment]
Thanks for your contribution.

This test really just tests initialization and a few lines of other code. It's probably still better than nothing. @fabxc What do you think?

File: discovery/consul/consul_test.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+)
+
+var configuredServiceName = "configuredService"
+var nonConfiguredServiceName = "nonConfiguredService"
+
+func TestConfiguredService(t *testing.T) {
+	conf := &config.ConsulSDConfig{
+		Services: []string{configuredServiceName}}
+
+	consulDiscovery, _ := NewDiscovery(conf)
```
Comment (fabxc) [inline discovery/consul/consul_test.go]: The error should definitely be checked as well. Same in other test.

File: discovery/consul/consul_test.go [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+package consul
+
+import (
+	"testing"
+
+	"github.com/prometheus/prometheus/config"
+)
+
+var configuredServiceName = "configuredService"
+var nonConfiguredServiceName = "nonConfiguredService"
```
Comment (fabxc) [inline discovery/consul/consul_test.go]: I think with the code is simple enough to just inline those in the tests.

**Comment by fabxc** [comment]
Certainly better than no tests :)
Added comments.

File: discovery/consul/consul_test.go (hunk 2 above)
Comment (neumayer) [inline discovery/consul/consul_test.go]: Done.

File: discovery/consul/consul_test.go (hunk 1 above)
Comment (neumayer) [inline discovery/consul/consul_test.go]: Done.

**Comment by neumayer** [comment]
I agree, they're not the most amazing tests ever :-).

I'll try to extend them in an extra pr when I find the time.

**Comment by fabxc** [comment]
Thanks 👍

**Pull request merged**